runtime-light/stdlib/file/file-system-functions.h

```diff
@@ -180,9 +188,38 @@ inline resource f$stream_socket_client(const string& address, std::optional<std:
 }
 
 inline Optional<string> f$file_get_contents(const string& stream) noexcept {
-  if (auto sync_resource{from_mixed<class_instance<kphp::fs::sync_resource>>(f$fopen(stream, {}), {})}; !sync_resource.is_null()) {
+  if (auto sync_resource{from_mixed<class_instance<kphp::fs::sync_resource>>(
+          f$fopen(stream, string{file_system_impl_::READ_MODE.data(), static_cast<string::size_type>(file_system_impl_::READ_MODE.size())}), {})};
+      !sync_resource.is_null()) {
     auto expected{sync_resource.get()->get_contents()};
     return expected ? Optional<string>{*std::move(expected)} : Optional<string>{false};
   }
   return false;
 }
+
+inline Optional<int64_t> f$file_put_contents(const string& stream, const mixed& content_var, int64_t flags = 0) noexcept {
+  string content{};
+  if (content_var.is_array()) {
+    content = f$implode(string(), content_var.to_array());
+  } else {
+    content = content_var.to_string();
+  }
+  std::span<const char> data_span{content.c_str(), content.size()};
+
+  constexpr int64_t FILE_APPEND_FLAG{1};
+  if (flags & ~FILE_APPEND_FLAG) {
+    kphp::log::warning("flags other, than FILE_APPEND are not supported in file_put_contents");
+    flags &= FILE_APPEND_FLAG;
+  }
+
+  const std::string_view& mode = ((flags & FILE_APPEND_FLAG) != 0) ? file_system_impl_::APPEND_MODE : file_system_impl_::WRITE_MODE;
+  if (auto sync_resource{
+          from_mixed<class_instance<kphp::fs::sync_resource>>(f$fopen(stream, string{mode.data(), static_cast<string::size_type>(mode.size())}), {})};
```

Comment:
Same comment as for `file_get_contents`: do we really need to create a general sync resource here instead of a file?

Reply:
It also works with `"php://stdout"` and `"php://stderr"`, so we really need sync resource.